### DISCUSSION TITLE
Sink and Gateway

### DISCUSSION THREAD

**Discussion by marcantoinebouchard** [mwaylabs/fruitymesh#208] [Q&A] [answered]
Hi, is there a way to send data directly to the sink? Does the node sending must know the sink id?

Thanks,
Marc

**Comment by mariusheil** [comment, marked as answer]
Hello,

you can use NodeId 31000 to send a message to your sink, you don't have to know its nodeId:
https://www.bluerange.io/docs/fruitymesh/Specification.html#NodeIds

Regards,
Marius

**Reply by marcantoinebouchard** [reply]
Thanks. Also, I read that there can only be one sink per network. Is that still the case? Any plans on implementing multiple sinks?

**Reply by mariusheil** [reply]
FruityMesh was designed in a way to support multiple sinks per network. The sink routing algorithm will determine which sink is the closest and will route messages on the direct path to that sink. You can try out this feature e.g. using CherrySim to set up a network of nodes with two sinks (as I quickly did in the screenshots below, sinks are circled red) and then query some of them about their hopsToSink by accessing one of the terminals usin "sim term ..." and checking this with the visualization:
![image](https://github.com/user-attachments/assets/cfb2af22-3143-4e22-93d1-aa35ebf7cc5d)
![image](https://github.com/user-attachments/assets/177922ba-ce45-4580-b4c6-da85c3b33c6a)

We have however had an issue with this feature in the past and have not yet tried to spot the exact cause as we did not have use-cases with multiple sinks for our main use case in building automation. If you spot a reproducible issue, we could certainly fix this. (Using the keep_alive messages will not work as this was not written with multiple sink nodes in mind and should be rewritten)

Depending on the traffic and topology, one could decide to seperate the network into multiple mesh networks (increases clustering time + reliability) or broadcast sensor messages to the network instead of routing them to the closest sink (slightly higher data usage in the network). This will work reliably in the current version and was tested in productive installations a lot.

Marius
